python/dlr/api.py

```diff
@@ -342,3 +343,41 @@ def get_input(self, name, shape=None):
                                      out.ctypes.data_as(ctypes.POINTER(ctypes.c_float))))
         out = out.reshape(shape)
         return out
+
+# Interface
+class IDLRModel:
+    __metaclass__=abc.ABCMeta
+
+    @abc.abstractmethod
+    def get_input_names(self):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_input(self, name, shape=None):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def run(self, input_data):
+        raise NotImplementedError
+
+# Wrapper class
+class DLRModel(IDLRModel):
+    def __init__(self, model_path, dev_type='cpu', dev_id=0):
+        #Determine if 3rdparty package needed
+        for (dirpath, dirnames, filenames) in os.walk(model_path):
+            for filename in filenames:
+                if filename.endswith('.pb'): 
+                    from .tf_model import TFModelImpl
+                    self._impl = TFModelImpl(model_path, dev_type, dev_id)
```

Comment:
if model_path is a folder then we should make sure the folder has only one `.pb` file.
BTW, I'd make support for both options:
1. for `.pb` file, e.g. `model_path=/tmp/mobilenet_v1/frozen_graph.pb`
2. for folder containing single pb file, e.g. `model_path=/tmp/mobilenet_v1`